floris/tools/layout_visualization.py

```diff
@@ -0,0 +1,579 @@
+            of bounding boxes (if show_bbox is True). Valid keys include:
+                * 'facecolor' (str):  Box background color. Defaults to 'gray'.
+                * 'alpha' (float): Opacity of box. Defaults to 0.5.
+                * 'pad' (float): Padding around text. Defaults to 0.1.
+                * 'boxstyle' (str): Box style (e.g., 'round'). Defaults to 'round'.
+        plotting_dict (Dict[str, Any], optional): Dictionary to control text
+            appearance. Valid keys include:
+                * 'color' (str): Text color. Defaults to 'black'.
+
+    Returns:
+        plt.Axes: The axes object used for the plot.
+
+    Raises:
+        IndexError: If any value in `turbine_indices` is an invalid turbine index.
+        ValueError: If the length of `turbine_names` does not match the number of turbines.
+    """
+
+    # Generate axis, if needed
+    if ax is None:
+        _, ax = plt.subplots()
+
+    # If turbine names not none, confirm has correct number of turbines
+    if turbine_names is not None:
+        if len(turbine_names) != len(fi.layout_x):
+            raise ValueError("Length of turbine_names not equal to number turbines in fi object")
+    else:
+        # Assign simple default numbering
+        turbine_names = [f"{i:03d}" for i in range(len(fi.layout_x))]
+
+    # If label_offset is None, use default value of r/8
+    if label_offset is None:
+        rotor_diameters = fi.floris.farm.rotor_diameters.flatten()
+        r = rotor_diameters[0] / 2.0
+        label_offset = r / 8.0
+
+    # If turbine_indices is not none, make sure all elements correspond to real indices
+    if turbine_indices is not None:
+        try:
+            fi.layout_x[turbine_indices]
+        except IndexError:
+            raise IndexError("turbine_indices does not correspond to turbine indices in fi")
+    else:
+        turbine_indices = list(range(len(fi.layout_x)))
+
+    # Generate plotting dictionary
+    default_plotting_dict = {
+        "color": "black",
+        "label": None,
+    }
+    plotting_dict = {**default_plotting_dict, **plotting_dict}
+
+    # If showing bbox is true, if bbox_dict is None, use a default
+    default_bbox_dict = {"facecolor": "gray", "alpha": 0.5, "pad": 0.1, "boxstyle": "round"}
+    bbox_dict = {**default_bbox_dict, **bbox_dict}
+
+    for ti in turbine_indices:
+        if not show_bbox:
+            ax.text(
+                fi.layout_x[ti] + label_offset,
+                fi.layout_y[ti] + label_offset,
+                turbine_names[ti],
+                **plotting_dict,
+            )
+        else:
+            ax.text(
+                fi.layout_x[ti] + label_offset,
+                fi.layout_y[ti] + label_offset,
+                turbine_names[ti],
+                bbox=bbox_dict,
+                **plotting_dict,
+            )
+
+    # Plot labels and aesthetics
+    ax.axis("equal")
+
+    return ax
+
+
+def plot_turbines_rotors(
```

Comment:
The other functions are "plot_turbine_xx" - singular turbine - so that might be good to be consistent

Reply:
Done